packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -971,75 +973,97 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
     }
 
     /**
-     * Checks that a given batch context has a time context which is below a given que element
-     * @param _context The batch context to validate has values lower.
-     * @param _queueIndex Index of the queue element we are validating came later than the context.
+     * For a fair of
+     * @param _earlierContext The batch context to validate has values lower.
+     * @param _laterContext The batch context whose time values should come AFTER all of _earlierContext's subsequent queue transactions
+     * @param _firstfirstSubsequentQueueIndex Index of the queue element we are validating came later than the _earlierContext
      * @param _queueRef The storage container for the queue.
      */
-    function _validateContextBeforeEnqueue(
-        BatchContext memory _context,
-        uint40 _queueIndex,
+    function _validateEnqueuesBetweenContexts(
+        BatchContext memory _earlierContext,
+        BatchContext memory _laterContext,
+        uint40 _firstfirstSubsequentQueueIndex,
```

Comment:
was this a typo?  This could be `earlierContextSubsequentQueueIndex` maybe?

Reply:
LOL wtf... yeah. will fix this.